### DISCUSSION TITLE
How to prevent repetitive motions?

### DISCUSSION THREAD

**Discussion by ceewanna** [roflcoopter/viseron#628] [General]
From what I observed, it seems while one motion hasn't really completed, the subsequent  motion event often got triggered. This results in a number of adjacent vdo files having repetitive motions. It seems while one motion is being recorded, there is another motion gets triggered and its recording gets started while the previous recording hasn't finished.

Btw, this is my motion config.
#
background_subtractor:
  motion_detector:
    cameras:
      cam1:
        threshold: 8
        alpha: 0.1
        area: 0.03
        fps: 2
        width: 640
        height: 360
        trigger_recorder: true
        recorder_keepalive: true
        max_recorder_keepalive: 60
      cam2:
        threshold: 8
        alpha: 0.1
        area: 0.03
        fps: 2
        width: 640
        height: 360
        trigger_recorder: true
        recorder_keepalive: true
        max_recorder_keepalive: 60
#

**Comment by roflcoopter** [comment]
The recordings should not overlap, because when one recording ends it deletes all the segments that belong to that recording. If your cameras have timestamps in the feed, do you see them overlapping between two recordings?

There is sadly no way to specify a grace period between recordings as of now

**Reply by ceewanna** [reply]
lookback and idle_timeout settings help minimize this overlap. Right now lookback is set at 1s and idle_timeout is 6s. I still see some slight overlap which I guess might come from the lookback of 1s. But overall, the whole thing looks great.

I observed the overlaps from scanning through the whole series of vdo files slightly faster than normal. I haven't got time to look at timestamps in details but I will explore this later. 

Thanks.

**Reply by ceewanna** [reply]
I have 2 files by the name.
rec/ch2/2023-07-18/160558.mp4
rec/ch2/2023-07-18/160609.mp4

The first file the time shown in the camera starts at 160558 as per the name of the file and ends at 160606. The second file is weird though. It starts 160552 and the video contains motion just right before what happens in the first file until 160557 and then it skips over to 160602 and it ends at 160621.

I have lookback: 1 and  idle_timeout: 6 for the camera.

**Reply by roflcoopter** [reply]
Skipping can occur if FFmpeg loses connection to the camera for a while and then has to restart, which causes gaps in the recordings unfortunately.
It shouldnt be possible to have overlapping recordings, i am in the process of re-doing that whole implementation while building 24/7 recordings. Will see if i can fix it

**Reply by ceewanna** [reply]
From the 2 sample files above, when the second file skipped to 160602, it overlapped for 4 seconds with the ending part of the first file which lasted until 160606.

That 24/7 is a good news to me. It would make the system self-contained and catch whatever missed out from motion detection. Right now I have to fall back to my commercial NVR for 24/7.

By the way, what do you plan to distinguish between motion and 24/7 videos? I have in mind that a suffix letter could be appended to the filename (date). For instance, M for motion G for general etc. I usually screen for events from the file list selection.
